Main.hs

```diff
@@ -1,4 +1,58 @@
 module Main where
 
+-- import Control.Exception
+
+import Control.Monad (void)
+import Data.Aeson (decode)
+import qualified Data.ByteString.Lazy as BSL
+import Data.Maybe (fromJust)
+import Data.Text
+import Kriti (runKriti)
+import Options.Applicative
+import System.IO (IOMode (ReadMode), openFile)
+
+data KritiOptions = KritiOptions
+  { _koJSONFile :: FilePath,
+    _koTemplateFile :: FilePath,
+    _koBindingSymbol :: Text
+  }
+  deriving (Show, Eq)
+
+-- TODO: we should add support for STDIN too
+kritiOpts :: Parser KritiOptions
+kritiOpts =
+  KritiOptions
+    <$> strOption
+      ( long "json" <> short 'j' <> metavar "JSON_FILE"
+          <> help "The JSON file to read"
+      )
+    <*> strOption
+      ( long "template" <> short 't' <> metavar "TEMPLATE_FILE"
+          <> help "The template file to use"
+      )
+    <*> strOption
+      ( long "bind" <> short 'b' <> metavar "BINDING_SYMBOL"
+          <> showDefault
+          <> value "$"
+          <> help "The symbol that's used to represent a JSON binding within the template"
+      )
+
 main :: IO ()
-main = putStrLn "Hello, Haskell!"
+main = runKritiInteractive =<< execParser opts
+  where
+    opts =
+      info (kritiOpts <**> helper) $
+        fullDesc
+          <> progDesc "transform JSON using the Kriti language"
+          <> header "kriti - a minimal JSON templating language based on Go's template language."
+
+runKritiInteractive :: KritiOptions -> IO ()
+runKritiInteractive (KritiOptions jsonFile templateFile rootSymbol) = do
+  void $ checkFilePath jsonFile
+  void $ checkFilePath templateFile
+
+  json <- BSL.readFile jsonFile
+  template <- readFile templateFile
+  print $ runKriti (pack template) [(rootSymbol, fromJust . decode $ json)]
```

Comment:
You should either case on the `Either` then `print` OR import `Data.Bitraversable` and use `bitraverse print print`. We should also serialize the aeson `Value` before printing it.

This way the success output will be actual serialized JSON data which could be piped into somethingelse.